### DISCUSSION TITLE
Empty display

### DISCUSSION THREAD

**Discussion by andsve** [sanni/cartreader#356] [Need help]
Hi, awesome project first of all!

I built the hardware recently (v4) but I’m not getting any output on the display. I’m no hardware genius so I’m fumbling in the dark a bit, sorry beforehand here since it will be a bit of open question… 🙂 Any suggestions of what could be wrong, or ideas what I should be investigating?

I have flashed the arduino with a quite up to date firmware (I’m on Mac however so had to combine the portable-zip stuff with the official IDE release but everything compiled and seemed to be loaded over correctly). I have the battery adapter but realised it could run without battery so soldered a switch and the corresponding cable according to the wiki for running without battery, but still nothing shown on the display… There is power to the display board though, I can se it “light up” (backlight or something) but no text (the knob light also lights up, first blue then red).

Thankful for any help or input!
/sven

**edit:** After some more testing tonight, I tried to use the serial monitor instead just to see if it gave me any more info. But the only thing it prints out is "SD error"... So, I'm guessing it must have issues communicating with the board with SD and display (ie the "Makerbase MKS Mini12864 v3.0" which I'm using). With that information, any ideas where I should be investigating? I tried switching the cables from/to the display+sd board and main board without any luck.

**Comment by sanni** [comment]
Have you tried without the clock generator addon PCB? Sometimes they arrive broken and make the Cart Reader stop working.

If this didn't help then most likely you got a short circuit or a cold solder joint somewhere on the SPI pins. The SD card and the LCD screen share those pins: D50, D51, D52, D53

![3wrqrtq3rqr3tr](https://user-images.githubusercontent.com/19169494/163731921-adfd3f96-7197-4df8-bbde-4e6ed0a93a55.jpg)
Probably best to re-flow those pins both on the Arduino Mega and on the Cart Reader PCB, they should also be labeled at the connector where the cables plug in.

It could also be that the Mini12864 module has a short circuit somewhere, so check all its pins with a magnifying glass too.

**Reply by andsve** [reply]
Hey thanks for the reply!

✔️  I have tried without the clock generator, no luck (tried this first, then tried it with it attached).
✔️ Re-soldered the SD+LCD pins, no luck either.
✔️ Also saw the section in the "Troubleshooting" page regarding investigating the LCD and SD parts separately. Sadly also without any luck...

❌ Lastly, which actually might be the real issue (and insanely stupid by me..), I re-verified the guide for soldering the Mega and saw that I de-soldered the wrong component regarding the SMD fuse! 😞 This was the first thing I soldered when I started on the project and haven't looked back on it since then... I removed the bottom component in this image (instead of the middle one as shown):
![image](https://user-images.githubusercontent.com/21482/164205492-ced35803-b3e9-42f1-951f-951bff7135f0.png)

I'm guessing this is the issue, but please correct me if I'm wrong. I have ordered a new Mega that will arrive next week, this time I will do it correctly! 😄

**Comment by sanni** [comment]
Have you tried removing the fuse and see if this fixes it?

**Reply by andsve** [reply]
I tried removing the middle fuse instead and solder the dupont wire to that solder point instead, but now the Arduino doesn't even light up (or show up in ports, in the Arduino IDE). 😞 I tried this a week ago and thought that I had simply messed up the Arduino and ordered a new one that arrived today...

This time I did it correctly, desoldered the mid fuse attached a dupont wire to the left pad (just as the picture above), but having the same behaviour as the last one where it doesn't seem to turn on at all...

Should the Arduino turn when "detached" from the cart reader PCB, while connected via USB cable? Because neither of mine seem to turn on anymore. 😞 Attaching a picture, but don't think it would really help. Any ideas or suggestions appreciated!

![mega](https://user-images.githubusercontent.com/21482/165141105-f16ce457-5821-40fb-830f-de5f2fab6de2.png)

**Comment by sanni** [comment]
Not turning on while not connected to the Cart Reader PCB is normal. We severed the Arduinos power by desoldering the fuse, it now gets power through the Cart Reader PCB. Basically we route the power from the Arduinos USB port to the Cart Reader PCB and back to the Arduino.
It also does not turn on when no cartridge slot adapter PCB is inserted into the main PCB.

**Comment by andsve** [comment]
Ah I see! I guess that makes sense now since the Arduino powered on when I previously removed the wrong fuse then. So the issue must be (at least highly likely) somewhere on my solder point on the PCB, since the screen still does not light up. 🤔 Should I expect the Arduino to light up when attached to the cart reader at least (if everything worked as expected)?

I soldered a new PCB just now, to make sure I didn't make any stupid mistakes on my first try. This time I also did the non-battery-version, ie attaching a switch and a wire as described in the wiki. But still nothing. Does the screen (ext1 and 2) need to be attached for it to start btw? 😞 

I have an extra display/SD I could try with as well perhaps... Sadly it's not an official Makerbase but a "Fysetc Mini 12864 Panel" so I'm not sure which pads I would need to short, or if it would even work.

Sorry for dragging this thread along, really appreciate all the help and the project as whole! ❤️
![IMG_0295](https://user-images.githubusercontent.com/21482/165152320-e5a13550-9ae9-4882-beb1-57892e1f0768.jpeg)
![IMG_0296](https://user-images.githubusercontent.com/21482/165152341-2b3b3c0b-aabc-44e0-8895-043f9ba906a4.jpeg)
![IMG_0297](https://user-images.githubusercontent.com/21482/165152345-bef0bd53-84c4-424c-bcfa-078363fbac80.jpeg)
![IMG_0298](https://user-images.githubusercontent.com/21482/165152348-039d5194-8aec-4e31-86c8-2d2af84019dc.jpeg)

**Comment by sanni** [comment]
You soldered the NES slot to the wrong side, backside instead of frontside, of the adapter PCB.

![IMG_1186](https://user-images.githubusercontent.com/19169494/165155529-ff61d8c4-0b34-4fb1-b6a6-8bbb6e1623c0.jpg)

**Reply by andsve** [reply]
Oh goddamn, ofc it is! 😄 Thank you so much! Gonna take a while to desolder and fix, wish me luck 😄
